src/features/commands.ts

```diff
@@ -65,23 +87,15 @@ export function dotnetRestoreAllProjects(server: OmnisharpServer) {
 
 	return serverUtils.requestWorkspaceInformation(server).then(info => {
 
-		let commands: Command[] = [];
+		if (!('DotNet in info') || info.DotNet.Projects.length < 1) {
+			return Promise.reject("No .NET Core projects found");
+		}
+		
+		let commandPromises = projectsToCommands(info.DotNet.Projects);
         
-        if ('DotNet' in info && info.DotNet.Projects.length > 0) {
-            for (let project of info.DotNet.Projects) {
-                commands.push({
-                    label: `dotnet restore - (${project.Name || path.basename(project.Path)})`,
-                    description: path.dirname(project.Path),
-                    execute() {
-                        return runInTerminal('dotnet', ['restore'], {
-                            cwd: path.dirname(project.Path)
-                        });
-                    }
-                });
-            }
-        }
-
-		return vscode.window.showQuickPick(commands).then(command => {
+		return Promise.all(commandPromises).then(commands => {
+			return vscode.window.showQuickPick(commands);
```

Comment:
I know you didn't change this but curious - why does it ask which project to restore instead of restoring all of them?